src/Factory/Form/MainSettingsFormTypeFactory.php

```diff
@@ -68,12 +68,21 @@ private function getInitialFormData(SettingsInterface $settings): array
         return $data + $this->getChannelInitialFormData($settings);
     }
 
+    /**
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
     private function getChannelInitialFormData(SettingsInterface $settings): array
     {
         $data = [];
         /** @var ChannelInterface $channel */
         foreach ($this->channelRepository->findAll() as $channel) {
-            $data['channel-' . $channel->getId() . '-' . Settings::DEFAULT_KEY] = $settings->getSettingsValuesByChannelAndLocale($channel);
+            $channelKey = 'channel-' . $channel->getId() . '-' . Settings::DEFAULT_KEY;
+            $data[$channelKey] = $settings->getSettingsValuesByChannelAndLocale($channel);
+            $defaultDataByChannel = $this->getDefaultValuesByChannel($settings, $channel);
+
+            if (isset($defaultDataByChannel[$channelKey])) {
+                $data[$channelKey] += $defaultDataByChannel[$channelKey];
```

Comment:
**🎯 suggestion:** ‏Are you sure about this?

What about this instead?

```suggestion
                $data[$channelKey][] = $defaultDataByChannel[$channelKey];
```

Reply:
Yeah, I'm sure, I want a union of the two arrays with the keys of `$data` if they exist in both array.
`$data[$channelKey][] =` will add me a new array on a numeric index which is not what I want